### DISCUSSION TITLE
Android app crashes on startup on tauri/tauri-build upgrade

### DISCUSSION THREAD

**Discussion by BSteffaniak** [tauri-apps/tauri#9927] [Q&A]
Hi, after upgrading from tauri 2.0.0-beta.20 to tauri 2.0.0-beta.21 and tauri-build 2.0.0-beta.16 to tauri-build 2.0.0-beta.17 I receive this error when starting the app and it crashes immediately:

```
...
Performing Streamed Install
Success
Starting: Intent { cmp=com.moosicbox_app/.MainActivity }
--------- beginning of main
05-30 15:38:36.843 22632 22632 I m.moosicbox_app: Late-enabling -Xcheck:jni
05-30 15:38:36.885 22632 22632 W m.moosicbox_app: Unexpected CPU variant for x86: x86_64.
05-30 15:38:36.885 22632 22632 W m.moosicbox_app: Known variants: atom, sandybridge, silvermont, kabylake, default
05-30 15:38:36.962 22632 22632 W ziparchive: Unable to open '/data/app/~~NKakGU_JYKht2BW4D-7BOA==/com.moosicbox_app-AcG2kaqCXxaxo0QLULrJyw==/base.dm': No such file or directory
05-30 15:38:36.962 22632 22632 W ziparchive: Unable to open '/data/app/~~NKakGU_JYKht2BW4D-7BOA==/com.moosicbox_app-AcG2kaqCXxaxo0QLULrJyw==/base.dm': No such file or directory
05-30 15:38:37.327 22632 22632 E m.moosicbox_app: No implementation found for void com.moosicbox_app.WryActivity.create(com.moosicbox_app.WryActivity) (tried Java_com_moosicbox_1app_WryActivity_create and Java_com_moosicbox_1app_WryActivity_create__Lcom_moosicbox_1app_WryActivity_2)
--------- beginning of crash
05-30 15:38:37.327 22632 22632 E AndroidRuntime: FATAL EXCEPTION: main
05-30 15:38:37.327 22632 22632 E AndroidRuntime: Process: com.moosicbox_app, PID: 22632
05-30 15:38:37.327 22632 22632 E AndroidRuntime: java.lang.UnsatisfiedLinkError: No implementation found for void com.moosicbox_app.WryActivity.create(com.moosicbox_app.WryActivity) (tried Java_com_moosicbox_1app_WryActivity_create and Java_com_moosicbox_1app_WryActivity_create__Lcom_moosicbox_1app_WryActivity_2)
05-30 15:38:37.327 22632 22632 E AndroidRuntime:        at com.moosicbox_app.WryActivity.create(Native Method)
05-30 15:38:37.327 22632 22632 E AndroidRuntime:        at com.moosicbox_app.WryActivity.onCreate(WryActivity.kt:62)
05-30 15:38:37.327 22632 22632 E AndroidRuntime:        at android.app.Activity.performCreate(Activity.java:8305)
05-30 15:38:37.327 22632 22632 E AndroidRuntime:        at android.app.Activity.performCreate(Activity.java:8284)
05-30 15:38:37.327 22632 22632 E AndroidRuntime:        at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1417)
05-30 15:38:37.327 22632 22632 E AndroidRuntime:        at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3626)
05-30 15:38:37.327 22632 22632 E AndroidRuntime:        at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3782)
05-30 15:38:37.327 22632 22632 E AndroidRuntime:        at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:101)
05-30 15:38:37.327 22632 22632 E AndroidRuntime:        at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:135)
05-30 15:38:37.327 22632 22632 E AndroidRuntime:        at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95)
05-30 15:38:37.327 22632 22632 E AndroidRuntime:        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2307)
05-30 15:38:37.327 22632 22632 E AndroidRuntime:        at android.os.Handler.dispatchMessage(Handler.java:106)
05-30 15:38:37.327 22632 22632 E AndroidRuntime:        at android.os.Looper.loopOnce(Looper.java:201)
05-30 15:38:37.327 22632 22632 E AndroidRuntime:        at android.os.Looper.loop(Looper.java:288)
05-30 15:38:37.327 22632 22632 E AndroidRuntime:        at android.app.ActivityThread.main(ActivityThread.java:7872)
05-30 15:38:37.327 22632 22632 E AndroidRuntime:        at java.lang.reflect.Method.invoke(Native Method)
05-30 15:38:37.327 22632 22632 E AndroidRuntime:        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:548)
05-30 15:38:37.327 22632 22632 E AndroidRuntime:        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:936)
        Info Watching C:\Users\Brade\GitHub\MoosicBoxApp\src-tauri for changes...
```

Maybe one of these PRs is causing this to now happen?

* https://github.com/tauri-apps/tauri/pull/9858
* https://github.com/tauri-apps/tauri/pull/9890

Am I missing an upgrade migration step?

Thanks!

**Comment by openersoft** [comment]
same problem.

**Reply by openersoft** [reply]
[✔] Environment
    - OS: Mac OS 14.4.1 X64
    ✔ Xcode Command Line Tools: installed
    ✔ rustc: 1.78.0 (9b00956e5 2024-04-29)
    ✔ cargo: 1.78.0 (54d8815d0 2024-03-26)
    ✔ rustup: 1.27.1 (54dd3d00f 2024-04-24)
    ✔ Rust toolchain: stable-aarch64-apple-darwin (environment override by RUSTUP_TOOLCHAIN)
    - node: 22.2.0
    - pnpm: 9.1.3
    - npm: 10.7.0

[-] Packages
    - tauri [RUST]: 2.0.0-beta.22 (no manifest)
    - tauri-build [RUST]: 2.0.0-beta.17 (no manifest)
    - wry [RUST]: 0.40.0 (no manifest)
    - tao [RUST]: 0.28.0 (no manifest)
    - tauri-cli [RUST]: 2.0.0-beta.9
    - @tauri-apps/api : not installed!
    - @tauri-apps/cli [NPM]: 2.0.0-beta.9

[-] App
    - build-type: bundle
    - CSP: unset
    - frontendDist: ../frontend/dist
    - devUrl: http://localhost:1420/

**Comment by BSteffaniak** [comment]
I am still getting this error with the latest versions of tauri and tauri-build, should we create an issue from this? @FabianLars

**Reply by FabianLars** [reply]
yes, ideally with a reproduction because my own app still works. Also, did you try to remove src-tauri/gen/android and rerun `tauri android init` ? (assuming you didn't modify the gen folder yourself yet)

**Reply by BSteffaniak** [reply]
Yes, I've tried re-initializing the android app gen and got the same error.

Ok, I'll try to create a minimal reproducible example repo to demonstrate the issue.

Thanks!

**Comment by zennolux** [comment]
Nearly the same issue when executing with `pnpm tauri android dev`

![image](https://github.com/tauri-apps/tauri/assets/147602513/b36a588b-7231-435c-b22e-273c262b3898)

**Reply by openersoft** [reply]
hi, would you problem solved? i'm stuckon this , still.

**Reply by openersoft** [reply]
fixed . upgrade create-tauri-app and rerun ‘cargo android init’ solved my problem.

**Comment by BSteffaniak** [comment]
Okay, I've been able to successfully create a stripped-down reproduction of this issue. I think the issue might be in one of the transitive dependencies being updated. Simply reverting the tauri and tauri-build versions did not fix the issue, I had to force a few other tauri package versions in the Cargo.lock.

Working on creating an issue now.

**Reply by BSteffaniak** [reply]
Issue created here: https://github.com/tauri-apps/tauri/issues/10024

**Comment by BSteffaniak** [comment]
This issue was caused by me having my @tauri-apps/cli version pinned (accidentally) to an old version. Removing that pinned version and updating to the latest version prompted me (through an error) to basically regenerate my android project (because the package name is handled differently now). Doing this has fixed this issue for me.

Thanks all for the help!
